internal/plugintest/working_dir.go

```diff
@@ -82,29 +82,71 @@ func (wd *WorkingDir) GetHelper() *Helper {
 // This must be called at least once before any call to Init, Plan, Apply, or
 // Destroy to establish the configuration. Any previously-set configuration is
 // discarded and any saved plan is cleared.
-func (wd *WorkingDir) SetConfig(ctx context.Context, cfg string) error {
+func (wd *WorkingDir) SetConfig(ctx context.Context, cfg teststep.Config, vars config.Variables) error {
+	// Remove old config and variables files first
+	d, err := os.Open(wd.baseDir)
+
+	if err != nil {
+		return err
+	}
+
+	defer d.Close()
+
+	fi, err := d.Readdir(-1)
+
+	if err != nil {
+		return err
+	}
+
+	for _, file := range fi {
+		if file.Mode().IsRegular() {
+			if filepath.Ext(file.Name()) == ".tf" || filepath.Ext(file.Name()) == ".json" {
```

Comment:
Open question: Not sure if we should or shouldn't be concerned about non-Terraform JSON files here.

Reply:
I _think_ that the only `.json` files within the `working_dir` will be those that are copied when a `.json` file, or a directory containing `.json` files is specified for `TestStep.ConfigFile`, or`TestStep.ConfigDirectory`, respectively. So I believe removing `.json` files will be ok from this perspective.

Do you know any way in which a non-Terraform JSON file might end-up in the working directory, and if so, whether removing it would be likely to cause issues?

Reply:
Hmm, I think you're right to a certain degree. Maybe I was thinking of the opposite problem then -- let's say we're using `ConfigDirectory` across multiple steps and I have:

```
# step 1 directory
|- file.whateverextension
 \ main.tf

# step 2 directory
\ main.tf
```

We should be preserving things necessary between steps in the working directory such as the `terraform.tfstate`, but removing the rest of the working directory files/directories. The configuration copying would overwrite any lingering files between steps, so in that regard its okay, but I think it would issue if the configuration was using something like `fileset()` instead of referencing thing with `file()` directory _and_ expecting those lingering files to be removed between steps. Let's treat this as something to follow up with rather than holding this up further though!

Reply:
Thanks for the additional context. Have created a follow-up issue - [Consider removal of all files copied from ConfigDirectory between sequential test steps](https://github.com/hashicorp/terraform-plugin-testing/issues/172#top)